Python/gc_free_threading.c

```diff
@@ -1952,25 +1953,25 @@ int
 PyGC_Enable(void)
 {
     GCState *gcstate = get_gc_state();
-    int old_state = gcstate->enabled;
-    gcstate->enabled = 1;
+    int old_state = _Py_atomic_load_int_relaxed(&gcstate->enabled);
+    while (!_Py_atomic_compare_exchange_int(&gcstate->enabled, &old_state, 1));
     return old_state;
 }
 
 int
 PyGC_Disable(void)
 {
     GCState *gcstate = get_gc_state();
-    int old_state = gcstate->enabled;
-    gcstate->enabled = 0;
+    int old_state = _Py_atomic_load_int_relaxed(&gcstate->enabled);
+    while (!_Py_atomic_compare_exchange_int(&gcstate->enabled, &old_state, 0));
```

Comment:
I think this can also just be an exchange, since we're unconditionally disabling GC:

```suggestion
    int old_state = _Py_atomic_exchange_int(&gcstate->enabled, 0);
```

Reply:
Yeah looks better, Thank you Matt :)